packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -522,219 +594,566 @@ class Navigator extends StatefulWidget {
+  ///    removed by reference to the route above it, rather than directly.
+  @optionalTypeArgs
+  static void replace<T extends Object>(BuildContext context, { @required Route<dynamic> oldRoute, @required Route<T> newRoute }) {
+    return Navigator.of(context).replace<T>(oldRoute: oldRoute, newRoute: newRoute);
   }
 
-  /// The state from the closest instance of this class that encloses the given context.
+  /// Replaces a route on the navigator that most tightly encloses the given
+  /// context with a new route. The route to be replaced is the one below the
+  /// given `anchorRoute`.
   ///
-  /// Typical usage is as follows:
+  /// {@template flutter.widgets.navigator.replaceRouteBelow}
+  /// The old route must not be current visible, as this method skips the
+  /// animations and therefore the removal would be jarring if it was visible.
+  /// To replace the top-most route, consider [pushReplacement] instead, which
+  /// _does_ animate the new route, and delays removing the old route until the
+  /// new route has finished animating.
   ///
-  /// ```dart
-  /// Navigator.of(context)
-  ///   ..pop()
-  ///   ..pop()
-  ///   ..pushNamed('/settings');
-  /// ```
+  /// The removed route is removed without being completed, so this method does
+  /// not take a return value argument.
   ///
-  /// If `rootNavigator` is set to true, the state from the furthest instance of
-  /// this class is given instead. Useful for pushing contents above all subsequent
-  /// instances of [Navigator].
-  static NavigatorState of(
-    BuildContext context, {
-      bool rootNavigator: false
-    }) {
-    final NavigatorState navigator = rootNavigator
-        ? context.rootAncestorStateOfType(const TypeMatcher<NavigatorState>())
-        : context.ancestorStateOfType(const TypeMatcher<NavigatorState>());
-    assert(() {
-      if (navigator == null) {
-        throw new FlutterError(
-          'Navigator operation requested with a context that does not include a Navigator.\n'
-          'The context used to push or pop routes from the Navigator must be that of a widget that is a descendant of a Navigator widget.'
-        );
-      }
-      return true;
-    }());
-    return navigator;
+  /// The new route, the route below the new route (if any), and the route above
+  /// the new route, are all notified (see [Route.didReplace],
+  /// [Route.didChangeNext], and [Route.didChangePrevious]). If the [Navigator]
+  /// has any [Navigator.observers], they will be notified as well (see
+  /// [NavigatorObservers.didReplace]). The removed route is disposed without
+  /// being notified. The future that had been returned from pushing that routes
+  /// will not complete.
+  ///
+  /// The `T` type argument is the type of the return value of the new route.
+  /// {@endtemplate}
+  ///
+  /// See also:
+  ///
+  ///  * [replace], which is the same but identifies the route to be removed
+  ///    directly.
+  @optionalTypeArgs
```

Comment:
would describe what would happen if anchorRoute is the root or is not in the stack etc

Reply:
Done.